drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -133,94 +140,126 @@ static void sdw_compute_master_ports(struct sdw_master_runtime *m_rt,
 		(*port_bo) += bps * ch;
 	}
 
+	t_data.lane = params->lane;
 	sdw_compute_slave_ports(m_rt, &t_data);
 }
 
 static void _sdw_compute_port_params(struct sdw_bus *bus,
 				     struct sdw_group_params *params, int count)
 {
 	struct sdw_master_runtime *m_rt;
-	int hstop = bus->params.col - 1;
-	int port_bo, i;
+	int port_bo, i, l;
+	int hstop;
 
 	/* Run loop for all groups to compute transport parameters */
-	for (i = 0; i < count; i++) {
-		port_bo = 1;
+	for (l = 0; l < SDW_MAX_LANES; l++) {
+		/* reset hstop for each lane */
```

Comment:
now I am not following this comment since we said in the previous patches that only the last stream will use a different allocation, so not sure why all lanes need to use a HSTOP that's reset if we don't touch them.

Reply:
> now I am not following this comment since we said in the previous patches that only the last stream will use a different allocation, so not sure why all lanes need to use a HSTOP that's reset if we don't touch them.

The hstop variable is not stored. We need to calculate the hstop for each port. And the hstop always starts with 'bus->params.col - 1' for each lane.